packages/liquidator/src/proxyTransactionWrapper.js

```diff
@@ -0,0 +1,288 @@
+      };
+
+      // Make sure to keep trying with this nonce
+      const nonce = await this.web3.eth.getTransactionCount(this.account);
+
+      // Min Gas Price, with a max gasPrice of x4
+      const minGasPrice = parseInt(this.gasEstimator.getCurrentFastPrice(), 10);
+      const maxGasPrice = 2 * 3 * minGasPrice;
+
+      // Doubles gasPrice every iteration
+      const gasPriceScalingFunction = ynatm.DOUBLES;
+
+      // Receipt without events
+      receipt = await ynatm.send({
+        sendTransactionFunction: gasPrice => liquidation.send({ ...txnConfig, nonce, gasPrice }),
+        minGasPrice,
+        maxGasPrice,
+        gasPriceScalingFunction,
+        delay: 60000 // Tries and doubles gasPrice every minute if tx hasn't gone through
+      });
+    } catch (error) {
+      return new Error("Failed to liquidate position🚨");
+    }
+
+    return {
+      type: "Standard EOA liquidation",
+      tx: receipt && receipt.transactionHash,
+      sponsor: receipt.events.LiquidationCreated.returnValues.sponsor,
+      liquidator: receipt.events.LiquidationCreated.returnValues.liquidator,
+      liquidationId: receipt.events.LiquidationCreated.returnValues.liquidationId,
+      tokensOutstanding: receipt.events.LiquidationCreated.returnValues.tokensOutstanding,
+      lockedCollateral: receipt.events.LiquidationCreated.returnValues.lockedCollateral,
+      liquidatedCollateral: receipt.events.LiquidationCreated.returnValues.liquidatedCollateral,
+      txnConfig
+    };
+  }
+
+  async _executeLiquidationWithDsProxy(liquidationArgs) {
+    const blockBeforeLiquidation = await this.web3.eth.getBlockNumber();
+
+    const reserveCurrencyLiquidator = new this.web3.eth.Contract(this.ReserveCurrencyLiquidator.abi);
+
+    // TODO: the liquidation args, as structured hare is hard to read and maintain. We should refactor the liquidation
+    // strategy to better pass around these parms as they are no longer directly fed into the liquidation method.
+    const callData = reserveCurrencyLiquidator.methods
+      .swapMintLiquidate(
+        this.uniswapRouterAddress, // uniswapRouter
+        this.financialContract._address, // financialContract
+        this.reserveToken._address, // reserveCurrency
+        liquidationArgs[0], // liquidatedSponsor
+        { rawValue: this.maxReserverTokenSpent }, // maxReserverTokenSpent
+        { rawValue: liquidationArgs[1].rawValue }, // minCollateralPerTokenLiquidated
+        { rawValue: liquidationArgs[2].rawValue }, // maxCollateralPerTokenLiquidated. This number need to be >= the token price.
+        { rawValue: liquidationArgs[3].rawValue }, // maxTokensToLiquidate. This is how many tokens the positions has (liquidated debt).
+        liquidationArgs[4]
+      )
+      .encodeABI();
+    const callCode = this.ReserveCurrencyLiquidator.bytecode;
+
+    const dsProxyCallReturn = await this.dsProxyManager.callFunctionOnNewlyDeployedLibrary(callCode, callData);
```

Comment:
note that this does not use `ynatm`. a separate PR should be added to implement this gas escalation within the `dsProxyManager`. I added a tasks to track this and will work on it as soon as this is done with code review: https://github.com/UMAprotocol/protocol/issues/2788

Reply:
Can we refactor some of this into a generalized function in  `common/src/ContractUtils`?

Reply:
yup! I will change the `dsProxyManager` to use this. 

However, it looks like your use of the `runTransaction` in the proposer does not use ynatm either. we should ideally modify the run transaction to use this so we consistently bump gas all over the place.

Reply:
I'm going to work on this in another PR